### PR TITLE
Update Assimp to version 4.1

### PR DESCRIPTION
**Pull request by jsigrist**
This PR contains the changes to use assimp v4.1 in runtimecore.

3rdparty vTest (from old 3rdparty repo): http://runtime-test.esri.com:8080/view/vTest/job/3rdparty_vtest_interface/288/downstreambuildview/
vTest (from old 3rdparty repo): http://runtime-test.esri.com:8080/view/vTest/job/vtest_interface/1797/downstreambuildview/

3rdparty vTest (current location): http://runtime-test.esri.com:8080/job/dev_3rdparty_vtest_interface/104/downstreambuildview/

### CODE REVIEW

File: assimp.lua [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_android.lua")
+
+    -- project specific configuration settings
+
+    configuration { "android*" }
+
+      defines {
```
Comment (chri7325) [inline assimp.lua]: This is duplicated above and can be removed.

File: revision.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#ifndef ASSIMP_REVISION_H_INC
+#define ASSIMP_REVISION_H_INC
+
+#define GitVersion 0xfcb571d
+#define GitBranch "jere9309/version4_1"
```
Comment (chri7325) [inline revision.h]: runtimecore?

File: assimp.lua [hunk 3]
```diff
@@ -364,6 +364,10 @@ project "assimp"
     -- configuration { "winuwp" } == _TARGET_IS_WINUWP
     -- -------------------------------------------------------------
 
+    defines {
```
Comment (chri7325) [inline assimp.lua]: This should go under the configuration block to safely go into winuwp

File: assimp.lua [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+  files {
+    "code/**.cpp",
+    "code/**.h",
+    "code/**.hpp",
+    "contrib/irrXML/irrXML.h", -- Required by Collada (DAE) reader
+    "contrib/irrXML/irrXML.cpp",
+    "contrib/rapidjson/include/**.h",
+  }
+
+  excludes {
```
Comment (chri7325) [inline assimp.lua]: nitpick: This can be removed

File: assimp.lua [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+  -- -------------------------------------------------------------
+  -- configurations
+  -- -------------------------------------------------------------
+
+  if (os.is("windows") and not _TARGET_IS_WINUWP) then
+    -- -------------------------------------------------------------
+    -- configuration { "windows" }
+    -- -------------------------------------------------------------
+
+    buildoptions {
```
Comment (chri7325) [inline assimp.lua]: nitpick: This block should actually go under the configuration to be scoped
```
configuration { "windows" }

  buildoptions {
```

**Comment by jsigrist** [comment]
@dwoodard-rtc I copied some changes from our current version of assimp to add unicode support; do you know if there's a different/better way after your character encoding work is done?

**Pull request merged**